elife/external-volume.sls

```diff
@@ -20,6 +20,20 @@ mount-point-external-volume:
     file.directory:
         - name: {{ pillar.elife.external_volume.directory }}
 
+mount-point-external-volume-existing-data-move-out:
+    cmd.run:
+        - name: |
+            touch {{ pillar.elife.external_volume.directory }}/ping
+            if test -e /etc/init.d/docker; then stop docker; status docker; fi
```

Comment:
even if an sysV init script exists there (like it does in 18.04 for some reason) it doesn't mean the stop/start/restart/status commands exist, which they don't in 18.04.

```
elife@heavybox--vagrant:~$ whereis stop
stop:
elife@heavybox--vagrant:~$ whereis status
status:
elife@heavybox--vagrant:~$ whereis start
start:
elife@heavybox--vagrant:~$ whereis restart
restart:
elife@heavybox--vagrant:~$ lsb_release -a
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 18.04.1 LTS
Release:	18.04
Codename:	bionic
```

Reply:
> IIRC stop $service is asynchronous so we need a polling loop?

depends on the application and, in systemd, the target - if we're affecting a state a service depends on, then the services are stopped and started together, asynchronously. if we're affecting a service directly then it waits for the app to stop until a timeout.

which machine is using 14.04 still (besides search?)

Reply:
Should not be a problem if they don't exist in 18.04 as the `systemctl` commands will also run. Assuming one of the two will always fail.